test/metrics/spanner-metrics-exporter.test.ts

```diff
@@ -0,0 +1,225 @@
+import {CloudMonitoringMetricsExporter} from '../../src/metrics/spanner-metrics-exporter';
+import {
+  SPANNER_METER_NAME,
+  METRIC_NAME_ATTEMPT_COUNT,
+  METRIC_NAME_ATTEMPT_LATENCIES,
+  METRIC_NAME_OPERATION_COUNT,
+  METRIC_NAME_OPERATION_LATENCIES,
+  METRIC_NAME_GFE_CONNECTIVITY_ERROR_COUNT,
+  METRIC_NAME_GFE_LATENCY,
+} from '../../src/metrics/constants';
+import {Counter, Meter, Histogram} from '@opentelemetry/api';
+import {ExportResult, ExportResultCode} from '@opentelemetry/core';
+
+const PROJECT_ID = 'test-project';
+const INSTANCE_ID = 'test_instance';
+const DATABASE_ID = 'test_db';
+
+// Ensure custom exporter is valid
+describe('CustomExporter', () => {
+  it('should construct an exporter', () => {
+    const exporter = new CloudMonitoringMetricsExporter();
+    assert.ok(typeof exporter.export === 'function');
+    assert.ok(typeof exporter.shutdown === 'function');
+  });
+
+  it('should construct an exporter with credentials', () => {
+    const exporter = new CloudMonitoringMetricsExporter({
+      authOptions: {
+        credentials: {
+          client_email: 'noreply@fake.example.com',
+          private_key: 'this is a key',
+        },
+      },
+    });
+
+    assert(exporter);
+    return (exporter['_projectId'] as Promise<string>).then(id => {
+      assert.deepStrictEqual(id, PROJECT_ID);
+    });
+  });
+
+  it('should be able to shutdown', async () => {
+    const exporter = new CloudMonitoringMetricsExporter();
+    await assert.doesNotReject(exporter.shutdown());
+  });
+
+  it('should be able to force flush', async () => {
+    const exporter = new CloudMonitoringMetricsExporter();
+    await assert.doesNotReject(exporter.forceFlush());
+  });
+});
+
+// Verify that the export call will convert and send the requests out.
+describe('Export', () => {
+  class InMemoryMetricReader extends MetricReader {
+    protected async onShutdown(): Promise<void> {}
+    protected async onForceFlush(): Promise<void> {}
+  }
+  let reader: MetricReader;
+  let meterProvider: MeterProvider;
+  let meter: Meter;
+  let attempt_counter: Counter;
+  let operation_counter: Counter;
+  let gfe_connectivity_error_count: Counter;
+  let attempt_latency: Histogram;
+  let operation_latency: Histogram;
+  let gfe_latencey: Histogram;
+  let metricAttributes: {[key: string]: string};
+  let exporter: CloudMonitoringMetricsExporter;
+
+  beforeEach(() => {
+    reader = new InMemoryMetricReader();
+    meterProvider = new MeterProvider({
+      readers: [reader],
+    });
+    meter = meterProvider.getMeter(SPANNER_METER_NAME);
+    metricAttributes = {
+      project_id: PROJECT_ID,
+      instance_id: INSTANCE_ID,
+      instance_config: 'test_config',
+      location: 'test_location',
+      client_hash: 'test_hash',
+      client_uid: 'test_uid',
+      client_name: 'test_name',
+      database: DATABASE_ID,
+      method: 'test_method',
+      status: 'test_status',
+      directpath_enabled: 'true',
```

Comment:
Please remove directpath reference from all the places

Reply:
Removed.